psi4/src/psi4/detci/opdm.cc

```diff
@@ -575,17 +576,61 @@ void CIWavefunction::opdm_block(struct stringwr **alplist, struct stringwr **bet
 void CIWavefunction::ci_nat_orbs() {
   outfile->Printf("\n   Computing CI Natural Orbitals\n");
 
+  // Grab orbital dimensions
+  Dimension zero_dim(nirrep_);
+  Dimension noccpi = get_dimension("DOCC");
+  Dimension nactpi = get_dimension("ACT");
+  Dimension nvirpi = get_dimension("VIR");
+
   // Diagonalize the OPDM in the active space
   SharedMatrix NO_vecs(new Matrix("OPDM Eigvecs", opdm_->rowspi(), opdm_->colspi()));
   SharedVector NO_occ(new Vector("OPDM Occuption", opdm_->rowspi()));
   opdm_->diagonalize(NO_vecs, NO_occ, descending);
 
-  // get a copy of the active orbitals
+  // get a copy of the active orbitals and rotate them
   SharedMatrix Cactv = get_orbitals("ACT");
   SharedMatrix Cnat = Matrix::doublet(Cactv, NO_vecs);
 
   // set the active block of Ca_
   set_orbitals("ACT",Cnat);
+
+  // semicanonicalize the DOCC and VIR blocks only for CASSCF
+  if (options_.get_str("WFN") == "CASSCF"){
```

Comment:
On second thought I do try to protect the CASSCF wavefunction can you add a `if(!somcscf_) throw ...`?